src/promptflow/tests/test_configs/flows/assistant_tool_v2/oai_assistant.py

```diff
@@ -0,0 +1,52 @@
+import asyncio
+import json
+
+from openai import AsyncOpenAI
+
+from promptflow import tool
+from promptflow.connections import OpenAIConnection
+from promptflow.contracts.types import AssistantDefinition, PromptTemplate
+from promptflow.executor._tool_invoker import DefaultToolInvoker
+
+
+@tool
+async def oai_assistant(conn: OpenAIConnection, content: PromptTemplate, assistant_id: str, assistant_definition: AssistantDefinition):
+    cli = AsyncOpenAI(api_key=conn.api_key, organization=conn.organization)
+    thread = await cli.beta.threads.create()
+    await cli.beta.threads.messages.create(
+        thread_id=thread.id,
+        role="user",
+        content=content,
+    )
+    invoker = DefaultToolInvoker.load_assistant_tools(assistant_definition.tools)
+    run = await cli.beta.threads.runs.create(
+        thread_id=thread.id, assistant_id=assistant_id,
+        instructions=assistant_definition.instructions,
+        tools=invoker.to_openai_tools()
+    )
+
+    outputs = []
+    while run.status != "completed":
```

Comment:
we probably use switch to handle all kinds of status, including failed.

Reply:
Will include in next pr.